config/config.go

```diff
@@ -361,6 +369,14 @@ func NewConfig() *Config {
 	flag.Var(cfg.CloneRoute, "clone-route", "clone all matching routes and replace filters and predicates of all matched routes")
 	flag.BoolVar(&cfg.WaitFirstRouteLoad, "wait-first-route-load", false, "prevent starting the listener before the first batch of routes were loaded")
 
+	// Forwarded headers
+	flag.Var(cfg.ForwardedHeadersList, "forwarded-headers", "comma separated list of headers to add to the incoming request before routing\n"+
+		"X-Forwarded-For sets or appends request remote IP to the X-Forwarded-For header\n"+
+		"X-Forwarded-For-prepend sets or prepends request remote IP to the X-Forwarded-For header, overrides X-Forwarded-For\n"+
```

Comment:
AFAIK RFC misses the detail about how to add another entry to the list either prepend or append. The majority (Nginx, Apache, Google LB, ..) does prepend and ALB is the only one that does append, so I would rather switch to x-forwarded-for-append and default to prepend.
Do you have other information?
Context: I looked at this, when I did the Source() predicate investigation, that resulted in SourceFromLast() predicate.

Reply:
The link https://en.wikipedia.org/wiki/X-Forwarded-For#Format says it is appended
> X-Forwarded-For: client, proxy1, proxy2
> where the value is a comma+space separated list of IP addresses, the left-most being the original client, and each successive proxy that passed the request adding the IP address where it received the request from. 

Nginx appends:
https://www.nginx.com/resources/wiki/start/topics/examples/forwarded/
> NGINX even provides a $proxy_add_x_forwarded_for variable to automatically append $remote_addr to any incoming X-Forwarded-For headers.

Apache docs are unclear
https://httpd.apache.org/docs/2.4/mod/mod_proxy.html
but this answer states it appends
https://stackoverflow.com/questions/40411974/what-determines-when-mod-proxy-only-sometimes-add-internal-ip-addresses-to-x-for
and the code appends
https://github.com/apache/httpd/blob/9e02e2000cc05754a9c1c9a9aead7ce2cc8ea6ee/modules/proxy/proxy_util.c#L3992-L3996
https://github.com/apache/apr/blob/6128ac5307a958dccb1af4ebbb5cfa95607a8454/tables/apr_tables.c#L720-L722

HAProxy appends
http://cbonte.github.io/haproxy-dconv/2.5/configuration.html#4-option%20forwardfor
> Since this
header is always appended at the end of the existing header list, the server
must be configured to always use the last occurrence of this header only.

Note also that HAProxy has `[ except <network> ]` option that does the same as the proposed `forwarded-headers-exclude-cidrs`

In fact I can not find any case where XFF would be prepended :man_shrugging:

Reply:
Google LB appends two values client ip and LB ip as far as I can tell https://cloud.google.com/load-balancing/docs/https#x-forwarded-for_header

Reply:
@AlexanderYastrebov you posted `X-Forwarded-For: client, proxy1, proxy2`, client is first (prepend), if google LB or any other proxy appends itself, it's not append (not prepend either, but what can we do). Of course the XFF header can not be trusted in case the proxy in front does not write the connecting client into a specified field in the header, because it can be faked otherwise by the client connecting to the proxy in front of us. 
Following the current skipper naming I would prefer to have prepend/append in both (following default filters) or do default/append (following Source vs. SourceFromLast predicate).

Reply:
> Following the current skipper naming I would prefer to have prepend/append in both (following default filters) or do default/append (following Source vs. SourceFromLast predicate).

I have removed `X-Forwarded-For-prepend` as I have not found any proxy implementation that _prepends_ request IP to the header value.